docs/onboarding_to_cluster.md

```diff
@@ -0,0 +1,151 @@
+
+In general we store all the cluster-scoped resources in a `cluster-scope` kustomize application within this repository.
+
+## Adding namespaces
+
+### Base resources
+
+Please run:
+
+```sh
+bash hack/onboarding.sh NAMESPACE_NAME OWNER_TEAM OPTIONAL_PROJECT_DESCRIPTION
+```
+
+This script will create
+
+- A namespace in `cluster-scope/base/namespaces/$NAMESPACE_NAME`
+- A blank user group for the `$OWNER_TEAM` if it does not exist yet in the `cluster-scope/base/groups/$OWNER_TEAM`
+- An RBAC component for the project admin role `RoleBinding` in `cluster-scope/components/namespace-admin-rolebinding/$OWNER_TEAM` and maps it to the newly added namespace.
+
+### Enabling namespace deployment to a specific cluster
+
+```sh
+cd cluster-scope/overlays/TARGET_CLUSTER
+kustomize edit add resource ../../base/namespaces/$NAMESPACE
+```
+
+## Adding users
+
+Please select target cluster in `cluster-scope/overlays/TARGET_CLUSTER` and change directory to this folder. There you either see a `secret-generator.yaml` (encrypted users) file or `group-user_patch.yaml` file (plaintext)
+
+### Encrypted overlay patch for users
+
+We'll use sops/ksops for the encryption in here
+
+#### Creating new group encrypted overlay patch
+
+If this is the first time your team is adding a user group, please ensure the `.sops.yaml` file in the repository root contains your team denotation. If it doesn't please add a section like this:
+
+```yaml
+# .sops.yaml
+creation_rules:
+  - encrypted_regex: "^users$"
+    path_regex: "cluster-scope/.*YOUR_TEAM_PREFIX.*"
+    pgp: "0508677DD04952D06A943D5B4DC4116D360E3276, YOUR_GPG_KEY_FINGERPRINT"
+  ...
+  # Make sure this group is below your changes! This rule would be applied if no other condition is met.
+  - encrypted_regex: "^users$"
+    pgp: "0508677DD04952D06A943D5B4DC4116D360E3276"
+  ...
+```
+
+Please create following resource describing the users in your group:
+
+```yaml
+# groups/GROUP_NAME.yaml
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+  name: GROUP_NAME
+  annotations:
+    kustomize.config.k8s.io/behavior: replace
+users:
+  - USER_1
+  - USER_2
+```
+
+Now please encrypt the file with sops and verify it contains a fingerprint of both GPG keys: the OperateFirst master key and your team's key.
+
+```sh
+$ sops -e groups/GROUP_NAME.yaml > groups/GROUP_NAME.enc.yaml
+$ grep 'fp: ' groups/GROUP_NAME.enc.yaml
+        fp: 0508677DD04952D06A943D5B4DC4116D360E3276
+        fp: YOUR_GPG_KEY_FINGERPRINT
+
+```
+
+Don't forget to remove the plaintext variant of the resource before staging for a commit:
+
+```sh
+rm groups/GROUP_NAME.yaml
+```
+
+And list the sops encrypted file in the secret generator file:
+
+```yaml
+# secret-generator.yaml
+files:
+---
+- groups/GROUP_NAME.enc.yaml
+```
+
+#### Adding user to encrypted overlay patch
+
+Use sops to edit the encrypted group resource patch as:
+
+```sh
+sops groups/GROUP_NAME.enc.yaml
+```
+
```

Comment:
Let's add another line here like: 

"Then simply append the list of users with the new users that need to be added."